django/db/models/query.py

```diff
@@ -820,7 +820,7 @@ def intersection(self, *other_qs):
     def difference(self, *other_qs):
         return self._combinator_query('difference', *other_qs)
 
-    def select_for_update(self, nowait=False, skip_locked=False):
+    def select_for_update(self, nowait=False, skip_locked=False, of_self=False):
```

Comment:
I thought `of` would be a list of paths to referenced tables with `self` being a special way to referenced the current table instead from the ticket description. So we could do something along the lines of:

e.g.
```python
Comment.objects.select_for_update(of={'author'})
```

Would result in
```
SELECT * FROM comment JOIN user ON (comment.user_id = user.id) FOR UPDATE OF user
```